extensions/ql-vscode/src/vscode-tests/no-workspace/query-results.test.ts

```diff
@@ -216,7 +219,81 @@ describe('query-results', () => {
       },
       sourceInfo as SourceInfo
     );
-    expect(results3).to.deep.eq({ a: 6, t: 'SarifInterpretationData' });
+    // We do not re-interpret if we are reading from a SARIF file. 
+    expect(spy).to.not.have.been.called;
+
+    expect(results3).to.have.property('t', 'SarifInterpretationData');
+    expect(results3).to.have.nested.property('runs[0].results');
+
+    // try a fourth time, but this time we use an invalid small SARIF file
+    spy.reset();
+    fs.writeFileSync(interpretedResultsPath, JSON.stringify({
+      a: '6' // Invalid: no runs or results
+    }), 'utf8');
+
+    await expect(
+      interpretResultsSarif(
+        mockServer,
+        metadata,
+        {
+          resultsPath, interpretedResultsPath
+        },
+        sourceInfo as SourceInfo)
+    ).to.be.rejectedWith('Parsing output of interpretation failed: Invalid SARIF file: expecting at least one run with result.');
+
+    // We do not attempt to re-interpret if we are reading from a SARIF file. 
+    expect(spy).to.not.have.been.called;
+
+    // Try a fifth time with a valid large SARIF file
+    spy.reset();
+    const validSarifStream = fs.createWriteStream(interpretedResultsPath, { flags: 'w' });
+
+    validSarifStream.write(JSON.stringify({
+      runs: [{ results: [] }] // A run needs results to succeed.
+    }), 'utf8');
+
+    for (let i = 0; i < 10000000; i++) {
+      validSarifStream.write(JSON.stringify({
+        a: '6'
+      }), 'utf8');
+    }
+    validSarifStream.end();
+
+    const results5 = await interpretResultsSarif(
+      mockServer,
+      metadata,
+      {
+        resultsPath, interpretedResultsPath
+      },
+      sourceInfo as SourceInfo
+    );
+    // We do not re-interpret if we are reading from a SARIF file. 
+    expect(spy).to.not.have.been.called;
+
+    expect(results5).to.have.property('t', 'SarifInterpretationData');
+    expect(results5).to.have.nested.property('runs[0].results');
+
+    // Try a sixth time with an invalid large SARIF file
+    spy.reset();
+    const invalidSarifStream = fs.createWriteStream(interpretedResultsPath, { flags: 'w' });
+    for (let i = 0; i < 10000000; i++) {
+      invalidSarifStream.write(JSON.stringify({
+        a: '6'
+      }), 'utf8');
+    }
+
+    await expect(
+      interpretResultsSarif(
+        mockServer,
+        metadata,
+        {
+          resultsPath, interpretedResultsPath
+        },
+        sourceInfo as SourceInfo)
+    ).to.be.rejectedWith('Parsing output of interpretation failed: Invalid SARIF file: expecting at least one run with result.');
```

Comment:
@aeisenberg: this test is failing locally for me and I was wondering if you could help me understand why. I'm writing to the filestream with the same loop as above. 

Instead of getting the error we expect about how we expect at least one run with result, we get the error

```
Parser cannot parse input: unexpected characters
```
which indicates that we hit this error prior to hitting EOF, where the error I've added in this PR would be thrown.

Also of note, when I decrease the number of iterations through the loop (even, say, to a single iteration), the error changes to:

```
Parser cannot parse input: expected a value
```

rather than the error we expect, even though the file we've written should be functionally the same as the file in the 4th test. This makes me think it's something to do with the write stream, but I'm surprised that it works fine for the last test (invalid large SARIF file). 

Let me know if you have any pointers on this ☝️ 😸

Reply:
I think I'm having success with using `fs.unlink()` to delete the previous large SARIF file. (Not sure why overwriting it with an option of `w` passed to the `WriteStream` wasn't doing the job, but I'll take it!)

Reply:
I think we also have `del` for this purpose, which may be more robust at deletion.

Reply:
Replaced with `del` 👍 (but it didn't fix the other large SARIF test problem unfortunately)